container-search/src/main/java/com/yahoo/search/grouping/request/GroupingOperation.java

```diff
@@ -194,6 +199,17 @@ public GroupingExpression getGroupBy() {
         return groupBy;
     }
 
+    /** Assigns an {@link FilterExpression} as the filter-by clause of this operation. */
+    @Beta
+    public GroupingOperation setFilterBy(FilterExpression exp) {
+        filterBy = Objects.requireNonNull(exp, "Filter expression can not be null");
+        return this;
+    }
+
+    /** Returns the {@link FilterExpression} assigned as the filter-by clause of this operation. */
+    @Beta
+    public Optional<FilterExpression> getFilterBy() { return Optional.ofNullable(filterBy); }
```

Comment:
is it really helpful to have just this one returning an Optional when all the other may return null?  I would prefer keeping the API consistently old-style here.

Reply:
I don't have any strong opinion - agreeing that consistency results in a less confusing API for users. I'll change it to return null on missing value.

Reply:
Removed use of `Optional` in [27efceb](https://github.com/vespa-engine/vespa/pull/33762/commits/27efceb9fe807364b701a67a97fdd25e17cf55de)